compiler-lib/src/lowering/mod.rs

```diff
@@ -0,0 +1 @@
+pub mod molki;
```

Comment:
I suspect this is not molki specific, but our lower end representation in general before registers are assigned, right? So why adopt this weird name as module name `molki` instead of e.g. `instruction_selection`. :]

`pub enum Instr` will probably grow very fast. so moving it to its own file my be beneficial.